docs/overview.rst

```diff
@@ -0,0 +1,154 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Overview
+========
+Invenio-Files-REST is a files storage module. It allows you to store and
+retrieve files in a similar way to Amazon S3 APIs. It provides a set of
+features:
+
+- An abstraction of physical files storage.
+- Configurable storage backends with the ability to build your very own.
+- A robust REST API.
+- Highly customizable access-control.
+- Secure file handling.
+- Integrity checking mechanism.
+- Support for large file uploads and multipart upload.
+- Signals for system events.
+
+The REST API follows best practices and supports, e.g.:
+
+- Content negotiation and links headers.
+- Cache control via ETags and Last-Modified headers.
+- Optimistic concurrency control via ETags.
+- Rate-limiting, Cross-Origin Resource Sharing, and various security headers.
+
+Here is an introduction of the main concepts.
+
+.. image:: _static/data_model.png
+    :align: center
+
+The physical layer
+------------------
+
+Provides physical access to files defining the storage locations and how to perform operations.
+
+Location
+++++++++
+A :code:`Location` is a representation of a storage system. A location is described by its :code:`name`
+and :code:`URI`. The URI could be a path in a local directory or a remote system.
+For example, a location could have as a name :code:`shared-folder` and URI  :code:`/mnt/shared`.
+
+Among all defined locations, one has to be set as the :code:`default`.
+
+You can learn how to use Locations in the section :ref:`usage-create-location`.
+
+Storage
++++++++
+A :code:`Storage` provides the interface to interact with a :code:`Location` and perform basic
+operations with files on it, such as retrieve, store or delete.
+
+By default, Invenio-Files-REST implements a storage for local files
+:py:class:`invenio_files_rest.storage.PyFSFileStorage`. It controls how files are physically stored,
+for example the folders and files structure.
+
+You can create new storage implementations and configure Invenio to use any existing storage.
+See the section :ref:`usage-storage-backends` to learn how to do it.
```

Comment:
I would rephrase it to something like: For more information on how to configure your own storage check out :ref:`usage-storage-backends` section.